src/test/java/com/vinylteam/vinyl/dao/jdbc/JdbcUniqueVinylDaoITest.java

```diff
@@ -268,4 +268,16 @@ void findManyByArtistEmptyTableTest() throws SQLException {
         assertTrue(actualUniqueVinyls.isEmpty());
     }
 
+    @Test
+    @DisplayName("Checks whether upsertOneUniqueVinyl set hasOffers = false for existing Vinyl")
+    void upsertOneUniqueVinylTest(){
+        UniqueVinyl vinyl = uniqueVinyls.get(2);
+        UniqueVinyl dbVinyl = uniqueVinylDao.findById(vinyl.getId());
+        assertTrue(dbVinyl.getHasOffers());
+        dbVinyl.setHasOffers(false);
+        uniqueVinylDao.upsertOneUniqueVinyl(dbVinyl);
+        UniqueVinyl dbVinylAfterUpdate = uniqueVinylDao.findById(vinyl.getId());
+        assertFalse(dbVinylAfterUpdate.getHasOffers());
```

Comment:
//prepare
UniqueVinyl expectedUniqueVinyl = dataGenerator.getUniqueVinylByNumber(3); - generated by template with hasOffers==false, so you don't need to use elements of the main list.
//when
(boolean actualIsUpdated = )uniqueVinylDao.upsert(expectedUniqueVinyl);
//then
(assertTrue(actualIsUpdated);)
List<UniqueVinyl> actualUniqueVinyls = dataFinder.findAllUniqueVinyls();
assertEquals(expectedUniqueVinyl, actualUniqueVinyls.get(2));

dont's:

- don't use elements of the main list.
- don't use other methods of the class you are testing.

do's:

- do enforce structure in unit tests: prepare - when - then.
- do use clear names "expected...", "actual..."|.